constraint-solver/src/quadratic_symbolic_expression.rs

```diff
@@ -252,13 +253,7 @@ impl<T: FieldElement, V: Ord + Clone + Hash + Eq> QuadraticSymbolicExpression<T,
             })
             .collect();
 
-        if !to_add
```

Comment:
If it is zero, the addition function will more or less do the same checks here and adding zero also has no effect on a SymbolicExpression.